integration/integration_test.go

```diff
@@ -6407,3 +6409,69 @@ func testListResourcesAcrossClusters(t *testing.T, suite *integrationTestSuite)
 		})
 	}
 }
+
+func testSFTP(t *testing.T, suite *integrationTestSuite) {
+	// Create Teleport instance.
+	teleport := suite.newTeleport(t, nil, true)
+	defer teleport.StopAll()
+
+	client, err := teleport.NewClient(helpers.ClientConfig{
+		Login:   suite.Me.Username,
+		Cluster: helpers.Site,
+		Host:    Host,
+	})
+	require.NoError(t, err)
+
+	// Create SFTP session.
+	ctx := context.Background()
+	proxyClient, err := client.ConnectToProxy(ctx)
+	require.NoError(t, err)
+
+	sftpClient, err := sftp.NewClient(proxyClient.Client.Client)
+	require.NoError(t, err)
+
+	// Create file that will be uploaded and downloaded.
+	tempDir := t.TempDir()
+	testFilePath := filepath.Join(tempDir, "testfile")
+	testFile, err := os.Create(testFilePath)
+	require.NoError(t, err)
+
+	_, err = testFile.WriteString("This is test data.")
+	require.NoError(t, err)
+	require.NoError(t, testFile.Sync())
+
+	// Test stat'ing a file.
+	fi, err := sftpClient.Stat(testFilePath)
+	require.NoError(t, err)
+	require.NotNil(t, fi)
+
+	// Test downloading a file.
+	testFileDownload := testFilePath + "-download"
+	downloadFile, err := os.Create(testFileDownload)
+	require.NoError(t, err)
+
+	remoteDownloadFile, err := sftpClient.Open(testFilePath)
+	require.NoError(t, err)
+
+	_, err = io.Copy(downloadFile, remoteDownloadFile)
+	require.NoError(t, err)
+
+	// Test uploading a file.
+	testFileUpload := testFilePath + "-upload"
+	remoteUploadFile, err := sftpClient.Create(testFileUpload)
+	require.NoError(t, err)
+
+	_, err = io.Copy(remoteUploadFile, testFile)
+	require.NoError(t, err)
+
+	require.NoError(t, remoteUploadFile.Close())
```

Comment:
Maybe put these inside `t.Cleanup()` actions so they happen even when the test fails.